test/k8sT/istio.go

```diff
@@ -316,12 +362,53 @@ var _ = SkipContextIf(func() bool {
 				Expect(err).To(BeNil(), "DNS entry is not ready after timeout")
 			}
 
-			By("Testing L7 filtering")
+			By("Testing Istio service configuration")
 			reviewsPodV1, err := kubectl.GetPods(helpers.DefaultNamespace, formatLabelArgument(app, reviews, version, v1)).Filter(podNameFilter)
 			Expect(err).Should(BeNil(), "Cannot get reviewsV1 pods")
+			ratingsPodV1, err := kubectl.GetPods(helpers.DefaultNamespace, formatLabelArgument(app, ratings, version, v1)).Filter(podNameFilter)
+			Expect(err).Should(BeNil(), "Cannot get ratingsV1 pods")
+			detailsPodV1, err := kubectl.GetPods(helpers.DefaultNamespace, formatLabelArgument(app, details, version, v1)).Filter(podNameFilter)
+			Expect(err).Should(BeNil(), "Cannot get detailsV1 pods")
 			productpagePodV1, err := kubectl.GetPods(helpers.DefaultNamespace, formatLabelArgument(app, productPage, version, v1)).Filter(podNameFilter)
 			Expect(err).Should(BeNil(), "Cannot get productpageV1 pods")
 
+			err = helpers.WithTimeout(func() bool {
+				allGood := true
+
+				allGood = shouldHaveService(reviewsPodV1.String(), ratings, apiPort, outbound) && allGood
+				allGood = shouldHaveService(ratingsPodV1.String(), ratings, apiPort, inbound) && allGood
+				allGood = shouldHaveService(productpagePodV1.String(), details, apiPort, outbound) && allGood
+				allGood = shouldHaveService(detailsPodV1.String(), details, apiPort, inbound) && allGood
+				allGood = shouldHaveService(productpagePodV1.String(), ratings, apiPort, outbound) && allGood
+
+				return allGood
+			}, "Istio sidecar proxies are not configured", &helpers.TimeoutConfig{Timeout: helpers.HelperTimeout})
```

Comment:
This should be an `Eventually`. We had a PR by Paul recently that converted `helpers.WithTimeout` and other variants to the native Ginkgo versions.

Reply:
Done.